src/Storages/StoragePostgreSQL.cpp

```diff
@@ -537,8 +544,11 @@ void registerStoragePostgreSQL(StorageFactory & factory)
 {
     factory.registerStorage("PostgreSQL", [](const StorageFactory::Arguments & args)
     {
-        auto configuration = StoragePostgreSQL::getConfiguration(args.engine_args, args.getLocalContext());
+        PostgreSQLSettings postgresql_settings; /// TODO: move some arguments from the arguments to the SETTINGS.
```

Comment:
looks like you are missing this code:
```
if (args.storage_def->settings)
    postgresql_settings.loadFromQuery(*args.storage_def);
```
then for settings for which we leave compatibility:
```
auto common_settings = args.getContext()->getSettingsRef();
if (!settings.connection_pool_size.changed)
    settings.connection_pool_size = common_settings.postgresql_connection_pool_size;
```